packages/@aws-cdk/aws-secretsmanager/test/secret.test.ts

```diff
@@ -425,6 +425,23 @@ test('secretValue', () => {
   });
 });
 
+describe('secretName', () => {
+  test.each([undefined, 'mySecret'])('when secretName is %s', (secretName) => {
+    const secret = new secretsmanager.Secret(stack, 'Secret', {
+      secretName,
+    });
+    new cdk.CfnOutput(stack, 'MySecretName', {
+      value: secret.secretName,
+    });
+
+    // Creates secret name by parsing ARN.
+    expect(stack).toHaveOutput({
+      outputName: 'MySecretName',
+      outputValue: { 'Fn::Select': [6, { 'Fn::Split': [':', { Ref: 'SecretA720EF05' }] }] },
```

Comment:
So, if I understand this correctly, this will return the name with the auto-generated suffix? So if I provide `seretName` as `'MySecret'` when constructing the, I'll get back `MySecret-123456`? Is it obvious to customers that this is what they'll get?

Reply:
> So, if I understand this correctly, this will return the name with the auto-generated suffix?

Yes.

> Is it obvious to customers that this is what they'll get?

I'm not sure, but it's all we can do, without doing something like resorting to creating a custom resource to parse the secret name. How would you suggest we increase the obvious-ness of this?

Reply:
One idea I have is, in the case the customer passed the `secretName` when creating the `Secret`, to return a Token that resolves to `secretName`, but adds the dependency between the two resources on resolution, kind of similar to how our automatic cross-stack references work (I understand this might be tricky to implement).

Basically, I'm worried that someone will use `secretName` somewhere where the name of the Secret is expected, and when it resolves to `MySecret-123456`, it will fail at deploy time.

@eladb want to pitch in here with your opinion, since you've reviewed a few of these last SecretsManager PRs?

Reply:
It seems totally fine to me, especially given the result is only resolved during deployment. 

As far as I understand, For all intents and purposes this _is_ the secret name and should be usable everywhere a secret name is expected. 

(The opposite is actually not always true - I.e there are places where only qualified names are allowed).

Reply:
For the record here, @skinny85  was absolutely correct here (+10 internet points!). SecretsManager actually *doesn't* accept the resource name as an identifier for DescribeSecret -- it's just about the only form of identifier they don't support. This is documented in their API and frankly, I should have caught this. SecretsManager is hard. :/

See #11736 for the fix.